src/modules/slack/utils.ts

```diff
@@ -1,218 +1,16 @@
-  if (!Buffer.isBuffer(request.payload)) {
-    throw Boom.internal('Payload is not a Buffer');
-  }
-
-  const body = request.payload.toString('utf-8');
-  const parsed = Querystring.parse(body) as unknown as SlackSlashCommandPayload;
-
-  const slashCommandPayload = slackSlashCommandPayloadSchema.validate(parsed, {
-    stripUnknown: true,
-  });
-
-  checkForSlackErrors(slashCommandPayload, parsed);
-
-  return slashCommandPayload.value;
-}
-
-export function parseSlackActionPayload(request: Request): Lifecycle.Method {
-  if (!Buffer.isBuffer(request.payload)) {
-    throw Boom.internal('Payload is not a Buffer');
-  }
-
-  const body = request.payload.toString('utf-8');
-  const { payload } = Querystring.parse(body);
-
-  const parsed:
-    | SlackActionsPayload
-    | SlackDialogsPayload
-    | SlackBlockKitPayload
-    | SlackShortcutPayload
-    | SlackDialogSubmissionPayload = JSON.parse(payload as string);
-
-  let mutableSlackPayload;
-  switch (parsed.type) {
-    case 'message_action':
-      mutableSlackPayload = slackActionsPayloadSchema.validate(parsed as SlackActionsPayload, {
-        stripUnknown: true,
-      });
-      break;
-    case 'block_actions':
-      mutableSlackPayload = slackBlockPayloadSchema.validate(parsed as SlackBlockKitPayload, {
-        stripUnknown: true,
-      });
-      break;
-    case 'shortcut':
-      mutableSlackPayload = slackSShortcutPayloadSchema.validate(parsed as SlackShortcutPayload, {
-        stripUnknown: true,
-      });
-      break;
-    case 'view_submission':
-      mutableSlackPayload = slackSSDialogSubmissionPayloadSchema.validate(
-        parsed as SlackDialogSubmissionPayload,
-        {
-          stripUnknown: true,
-        }
-      );
-      break;
-    default:
-      const { view }: SlackActionsPayload = parsed as SlackActionsPayload;
-      mutableSlackPayload =
-        view && view.callback_id === 'modal-appreciation'
-          ? slackActionsPayloadSchema.validate(parsed as SlackActionsPayload, {
-              stripUnknown: true,
-            })
-          : slackDialogsPayloadSchema.validate(parsed as SlackDialogsPayload, {
-              stripUnknown: true,
-            });
-      break;
-  }
-  checkForSlackErrors(mutableSlackPayload, parsed);
-  return mutableSlackPayload.value;
-}
-
-export const parseSlackEventPayload: Lifecycle.Method = (request) => {
-  if (!Buffer.isBuffer(request.payload)) {
-    throw Boom.internal('Payload is not a Buffer');
-  }
-
-  const body = request.payload.toString('utf-8');
-  const { payload } = Querystring.parse(body);
-
-  const parsed: SlackChallengesPayload | SlackEventsPayload = JSON.parse(payload as string);
-
-  const slackEventPayload = parsed.challenge
-    ? slackChallengesPayloadSchema.validate(parsed, { stripUnknown: true })
-    : slackEventsPayloadSchema.validate(parsed, { stripUnknown: true });
-
-  checkForSlackErrors(slackEventPayload, parsed);
-
-  return slackEventPayload.value;
-};
```

Comment:
the refactor was basically focused on shortening file size and attempting to keep each files with less responsibilities. This will also help create less tests for each file